### PR TITLE
JobStartDate may be eventually missing in the classAd

### PR DESCRIPTION
**Pull request by amaltaro**
The component crashed a couple of days ago, because that ad was missing in the Ads.
At some point we should review this plugin, it has way too many possible states and assumptions flying over the code :(

@ticoann don't merge yet, I wan't to make sure no more crashes will show up.

### CODE REVIEW

**Comment by amaltaro** [comment]
Did some rework on this code mainly to make it more "readable", some improvements and also fixed a bug in the updateSiteInformation method.

I don't really have time to look at the log stuff right now, but will try to come back to it in the coming weeks.

Needs test, don't merge please.

File: src/python/WMCore/BossAir/Plugins/PyCondorPlugin.py [hunk 1]
```diff
@@ -343,6 +356,8 @@ def submit(self, jobs, info=None):
         # If we're here, then we have submitter components
         self.scriptFile = self.config.JobSubmitter.submitScript
         self.submitDir = self.config.JobSubmitter.submitDir
+        logging.info("ALAN self.submitDir %s and self.config.JobSubmitter.submitDir %s", self.submitDir,
```
Comment (amaltaro) [inline src/python/WMCore/BossAir/Plugins/PyCondorPlugin.py]: hmm, and of course I need to remove this line before it gets merged :)

Comment (ticoann) [inline src/python/WMCore/BossAir/Plugins/PyCondorPlugin.py]: I think you can keep it. :smile:

Comment (amaltaro) [inline src/python/WMCore/BossAir/Plugins/PyCondorPlugin.py]: LoL. I couldn't find where we set this var (and it's not in the config), that's why I wanted to check on it :-)

**Comment by amaltaro** [comment]
@ticoann good to go

**Pull request merged**